ocs_ci/ocs/ui/base_ui.py

```diff
@@ -78,44 +78,67 @@ def __init__(self, driver: WebDriver):
+        refresh the page once and try click again
+        * don't use when refreshed page expected to be different from initial page, or loose input values
         """
-        if (
-            version.get_semantic_version(get_ocp_version(), True)
-            <= version.VERSION_4_11
-        ):
-            try:
-                wait = WebDriverWait(self.driver, timeout)
+
+        def _do_click(_locator, _timeout=30, _enable_screenshot=False, _copy_dom=False):
+            if (
+                version.get_semantic_version(get_ocp_version(), True)
+                <= version.VERSION_4_11
+            ):
+                try:
+                    wait = WebDriverWait(self.driver, _timeout)
+                    element = wait.until(
+                        ec.element_to_be_clickable((_locator[1], _locator[0]))
+                    )
+                    screenshot = (
+                        ocsci_config.UI_SELENIUM.get("screenshot")
+                        and _enable_screenshot
+                    )
+                    if screenshot:
+                        self.take_screenshot()
+                    element.click()
+                    if _copy_dom:
+                        self.copy_dom()
+                except TimeoutException as e:
+                    self.take_screenshot()
+                    self.copy_dom()
+                    logger.error(e)
+                    raise TimeoutException
+            else:
+                self.page_has_loaded()
+                wait = WebDriverWait(self.driver, _timeout)
                 element = wait.until(
-                    ec.element_to_be_clickable((locator[1], locator[0]))
-                )
-                screenshot = (
-                    ocsci_config.UI_SELENIUM.get("screenshot") and enable_screenshot
+                    ec.visibility_of_element_located((_locator[1], _locator[0]))
                 )
-                if screenshot:
-                    self.take_screenshot()
                 element.click()
-                if copy_dom:
-                    self.copy_dom()
-            except TimeoutException as e:
-                self.take_screenshot()
-                self.copy_dom()
-                logger.error(e)
-                raise TimeoutException
-        else:
-            self.page_has_loaded()
-            wait = WebDriverWait(self.driver, timeout)
-            element = wait.until(
-                ec.visibility_of_element_located((locator[1], locator[0]))
-            )
-            element.click()
+
+        try:
+            _do_click(locator, timeout, enable_screenshot, copy_dom)
+        except StaleElementReferenceException:
+            if avoid_stale:
+                logger.info("Refresh page to avoid reference to a stale element")
+                self.driver.refresh()
+                _do_click(locator, timeout, enable_screenshot, copy_dom)
+            else:
+                raise
```

Comment:
Any specific exception to be raised? Currently which exception is being raised here?

Reply:
raise itself will raise the same exception we got. This way we do refresh page and try _do_click once more, but this time do not pass avoid_stale and don't go into a new recursion and raise initial exception